File: year2/systems-design-implementation/Lab10/SwimServer/src/main/java/repo/intf/Repository.java
package repo.intf;

import utils.Entity;

import java.util.List;

public interface Repository<ID, E> {

    E findOne(ID id) throws Exception;

    List<E> findAll();

    E add(E entity);

    E delete(ID id) throws Exception;

    E update(E entity) throws Exception;

}

File: year2/systems-design-implementation/Lab8/SwimService/Repository/Interface/Repository.cs
using SwimModel.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SwimServer.Repository.Interface
{
    public interface Repository<ID, E> where E : Entity<ID>
    {
        E FindOne(ID id);
        IEnumerable<E> FindAll();
        E Add(E entity);
        E Delete(ID id);
        E Update(E entity);
    }
}
